feature-libs/product/configuration/src/textfield-configurator/facade/configurator-textfield.service.ts

```diff
@@ -0,0 +1,198 @@
+   */
+  createConfiguration(
+    owner: GenericConfigurator.Owner
+  ): Observable<ConfiguratorTextfield.Configuration> {
+    return this.store.pipe(
+      select(ConfiguratorTextFieldSelectors.getConfigurationsState),
+      tap((configurationState) => {
+        const isAvailableForProduct =
+          configurationState.loaderState.value?.owner.type ===
+          GenericConfigurator.OwnerType.PRODUCT;
+        const isLoading = configurationState.loaderState.loading;
+        if (!isAvailableForProduct && !isLoading) {
+          this.store.dispatch(
+            new ConfiguratorTextfieldActions.CreateConfiguration({
+              productCode: owner.id, //owner Id is the product code in this case
+              owner: owner,
+            })
+          );
+        }
+      }),
+      map((configurationState) => configurationState.loaderState.value),
+      filter((configuration) => !this.isConfigurationInitial(configuration))
+    );
+  }
+
+  /**
+   * Updates a textfield configuration, specified by the changed attribute.
+   *
+   * @param changedAttribute - Changed attribute
+   */
+  updateConfiguration(
+    changedAttribute: ConfiguratorTextfield.ConfigurationInfo
+  ): void {
+    this.store
+      .pipe(
+        select(ConfiguratorTextFieldSelectors.getConfigurationContent),
+        take(1)
+      )
+      .subscribe((oldConfiguration) => {
+        this.store.dispatch(
+          new ConfiguratorTextfieldActions.UpdateConfiguration(
+            this.createNewConfigurationWithChange(
+              changedAttribute,
+              oldConfiguration
+            )
+          )
+        );
+      });
+  }
+
+  /**
+   * Adds the textfield configuration to the cart
+   *
+   * @param productCode - Product code of the configuration root product. Cart entry carries refers to this product
+   * @param configuration Textfield configuration
+   */
+  addToCart(
+    productCode: string,
+    configuration: ConfiguratorTextfield.Configuration
+  ): void {
+    this.activeCartService
+      .requireLoadedCart()
+      .pipe(take(1))
+      .subscribe((cartState) => {
+        const addToCartParameters: ConfiguratorTextfield.AddToCartParameters = {
+          userId: this.configuratorUtils.getUserId(cartState.value),
+          cartId: this.configuratorUtils.getCartId(cartState.value),
+          productCode: productCode,
+          configuration: configuration,
+          quantity: 1,
+        };
+        this.store.dispatch(
+          new ConfiguratorTextfieldActions.AddToCart(addToCartParameters)
+        );
+      });
+  }
+
+  /**
+   * Updates a cart entry, specified by its cart entry number.
+   *
+   * @param cartEntryNumber - Cart entry number
+   * @param configuration Textfield configuration (list of alphanumeric attributes)
+   */
+  updateCartEntry(
+    cartEntryNumber: string,
+    configuration: ConfiguratorTextfield.Configuration
+  ): void {
+    this.activeCartService.requireLoadedCart().subscribe((cartState) => {
```

Comment:
Can we add `take(1)` in the pipe here, just in case the internals of `requireLoadedCart` change?

Reply:
Sorry forgot

Reply:
There is one more place where we can add it